app/controllers/comments_controller.rb

```diff
@@ -1,19 +1,48 @@
 class CommentsController < ApplicationController
-  before_action :comment, only: %i[show]
-
+  # GET /posts/:post_id/comments or /comments/:comment_id/comments
   def index
-    @comments = policy_scope(Comment).order(created_at: :desc)
-    render json: @comments.map { |comment| CommentSerializer.call(comment) }
+       post_comments = policy_scope(Comment).where(commentable_type: 'Post').order(created_at: :desc)
+       comment_replies = policy_scope(Comment).where(commentable_type: 'Comment').order(created_at: :desc)
```

Comment:
Porquê as respostas de um post ou de um comment não são carregadas pelo serializer?

Reply:
Só toma cuidado pra fazer um `.includes(:comments)` ou algo proximo a isso quando for carregar a collection da index pra evitar queries N+1